funnel/views/mixins.py

```diff
@@ -130,3 +133,75 @@ def loader(self, profile, project, section):
             ).first_or_404()
         g.profile = section.project.profile
         return section
+
+
+class DraftModelViewMixin(object):
+    def get_draft(self, current_obj=None):
+        """
+        Returns the draft object for `current_obj`. Defaults to `self.obj`.
+        `current_obj` is needed in case of multi-model views.
+        """
+        obj = current_obj or self.obj
```

Comment:
`obj = current_obj if current_obj is not None else self.obj`. An object may intentionally represent itself as a boolean (implementing `__bool__` or `__nonzero__`), so this is not a safe assumption when handling objects of unknown type.

Reply:
Also, make the parameter just `obj`.